### PR TITLE
feat(histogram): Adding support for min max for prom and delta histograms

### PR DESCRIPTION
**Pull request by sandeep6189**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [x] Docs have been added / updated (for bug fixes / features) ?

**New behavior :** Adding support for min max for prom and delta histograms

### CODE REVIEW

File: core/src/main/resources/filodb-defaults.conf [hunk 1]
```diff
@@ -183,6 +183,33 @@ filodb {
...
+      # Downsample periods are determined by counter dips of the count column
+      downsample-period-marker = "counter(2)"
+      downsample-schema = "prom-histogram-min-max"
+    }
+
+    delta-histogram-min-max {
+      columns = ["timestamp:ts",
+        "sum:double:{detectDrops=false,delta=true}",
+        "count:double:{detectDrops=false,delta=true}",
+        "min:double:{detectDrops=false,delta=true}",
```
Comment (amolnayak311) [inline core/src/main/resources/filodb-defaults.conf:204]: Based on [this](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#histogram) page the min and max fields are optional. For delta this especially means the value for min field can now be negative unlike sum, count and max.
Also, how should we interpret the word “optional”.  Is it guaranteed that once min and max are published they will always be published on each interval (even if same value or delta of 0) or the lack of this field be treated as delta of 0 ( I’m assuming this might be done to minimize data sent over the wire from agents?) in our case these are different schemas and we don’t want to fail queues with multiple schemas found error.

Comment (sandeep6189) [reply core/src/main/resources/filodb-defaults.conf:204]: Good question @amolnayak311 . From what I understand, we are adding a different schema altogether, which *needs* to have min-max provided by the agent (we are creating a new type/proto altogether to avoid this confusion) and the gateway would then consistently create the required schema, avoiding this issue altogether.

@sherali42

File: core/src/main/resources/filodb-defaults.conf [hunk 2]
```diff
@@ -183,6 +183,33 @@ filodb {
       downsample-schema = "delta-histogram"
     }
 
+    prom-histogram-min-max {
```
Comment (amolnayak311) [inline core/src/main/resources/filodb-defaults.conf:186]: Do you want to call it prom anymore or something like otel? Technically Prometheus histograms dont have min max and prom stands for Prometheus.

Comment (sandeep6189) [reply core/src/main/resources/filodb-defaults.conf:186]: I like the idea of calling it `otel_histogram`.

File: core/src/main/scala/filodb.core/metadata/Dataset.scala [hunk 3]
```diff
@@ -115,7 +115,7 @@ object DatasetOptions {
                                       metricColumn = "__name__",
                                       // defaults that work well for Prometheus
                                       ignoreShardKeyColumnSuffixes =
-                                        Map("__name__" -> Seq("_bucket", "_count", "_sum")),
+                                        Map("__name__" -> Seq("_bucket", "_count", "_sum", "_min", "_max")),
```
Comment (sherali42) [inline core/src/main/scala/filodb.core/metadata/Dataset.scala:118]: @sandeep6189 this is not needed, we can revert this change

Comment (sandeep6189) [reply core/src/main/scala/filodb.core/metadata/Dataset.scala]: Ack, reverting

File: core/src/main/resources/filodb-defaults.conf (hunk 2 above)
Comment (sandeep6189) [reply core/src/main/resources/filodb-defaults.conf]: updated to Otel-histogram

**Review by sherali42 — APPROVED** [review]
thanks @sandeep6189 for adding support for min/max histograms
